campaign-launcher/client/src/components/JoinCampaign/index.tsx

```diff
@@ -38,15 +43,31 @@ const JoinCampaign: FC<Props> = ({
     campaign.status === CampaignStatus.TO_CANCEL ||
     campaign.end_date < new Date().toISOString();
 
-  const isLoading = isEnrolledExchangesLoading || isJoinedLoading || isJoining;
+  const isLoading =
+    isEnrolledExchangesLoading || isJoinStatusLoading || isJoining;
+
+  const isAlreadyJoined = joinStatus === CampaignJoinStatus.USER_ALREADY_JOINED;
+  const isJoinLimited = joinStatus === CampaignJoinStatus.JOIN_IS_LIMITED;
+
   const isButtonDisabled =
-    !isAuthenticated || isLoading || isAlreadyJoined || isCampaignFinished;
+    !isAuthenticated ||
+    isLoading ||
+    !joinStatus ||
+    isAlreadyJoined ||
+    isJoinLimited ||
+    isCampaignFinished;
```

Comment:
i believe `isCampaignFinished` can be removed, as we return null in that case (line 103)

Reply:
@dnechay in order to cover the error case we can add `!joinStatus` to the statement, that returns null for the whole component